src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            return new ArrayList<>(new BulldozerProducer().get());
+        } else if (type.equals(Excavator.class)) {
+            return new ArrayList<>(new ExcavatorProducer().get());
```

Comment:
we don't need else after return in if 
```suggestion
        } 
         if (type.equals(Excavator.class)) {
            return new ArrayList<>(new ExcavatorProducer().get());
```